File: 1-9/2/main.cpp
#include <iostream>

int main() {
    long long sum = 0;
    long long current = 1;
    long long previous = 0;

    while (current <= 4000000) {
        if (current % 2 == 0) sum += current;

        long long next = current + previous;
        previous = current;
        current = next;
    }

    std::cout << "Sum of even Fibonacci numbers: " << sum << std::endl;

    return 0;
}

File: 1-9/2/main.go
package main

func main() {
	sum := 0
	current := 1
	previous := 0
	temp := 0

	for {
		if current > 4000000 {
			break
		}

		if current%2 == 0 {
			sum += current
		}

		temp = current
		current = current + previous
		previous = temp
	}

	println("Sum of even Fibonacci numbers:", sum)
}
